app/client/components/Watch.tsx

```diff
@@ -105,7 +120,13 @@ export class Watch extends React.Component<WatchProps, WatchState> {
     }
 
     private mkVideo(props): Promise<Video> {
-        return Video.readFromStringXorName(props.match.params.xorName);
+        const match: string[] = /\/watch\/([a-zA-Z0-9\/\+]+)/.exec(props.location.pathname);
+        if (match.length !== 2) {
+            this.setState({ badXorName: true });
```

Comment:
This setState call inside a setState call worries me.  Technically setState is async, and so I'm not convinced the order of events is always well defined